Source/EventAnalysis/KTOverlappingTrackClustering.hh

```diff
@@ -56,46 +70,191 @@ namespace Katydid
             virtual ~KTOverlappingTrackClustering();
 
             bool Configure(const scarab::param_node* node);
+            bool TakeTrack(KTProcessedTrackData& track);
+            bool TakeSeqLineCandidate(KTSequentialLineData& swfCand);
+            bool Run();
+            const std::set< Nymph::KTDataPtr >& GetCandidates() const;
+
 
+        private:
             MEMBERVARIABLE(double, MaxTrackWidth);
-            MEMBERVARIABLE(bool, ApplyPowerCut);
-            MEMBERVARIABLE(bool, ApplyDensityCut);
-            MEMBERVARIABLE(double, PowerThreshold);
-            MEMBERVARIABLE(double, DensityThreshold);
+            MEMBERVARIABLE(double, LargeMaxTrackWidth);
+            MEMBERVARIABLE(bool, ApplyTotalPowerCut);
+            MEMBERVARIABLE(bool, ApplyAveragePowerCut);
+            MEMBERVARIABLE(bool, ApplyTotalSNRCut);
+            MEMBERVARIABLE(bool, ApplyAverageSNRCut);
+            MEMBERVARIABLE(bool, ApplyTotalUnitlessResidualCut);
+            MEMBERVARIABLE(bool, ApplyAverageUnitlessResidualCut);
+            MEMBERVARIABLE(double, TotalPowerThreshold);
+            MEMBERVARIABLE(double, AveragePowerThreshold);
+            MEMBERVARIABLE(double, TotalSNRThreshold);
+            MEMBERVARIABLE(double, AverageSNRThreshold);
+            MEMBERVARIABLE(double, TotalUnitlessResidualThreshold);
+            MEMBERVARIABLE(double, AverageUnitlessResidualThreshold);
             MEMBERVARIABLE(unsigned, NTracks);
 
 
-        public:
-            // Store point information locally
-            bool TakeTrack(KTProcessedTrackData& track);
+        private:
+            template<typename T>
+            bool DoCandidateClustering(std::vector<T>& compCands, std::vector<T>& newCands)
+            {
+                if (! FindMatchingCandidates(compCands, newCands))
+                {
+                    KTERROR(otchlog, "An error occurred while identifying overlapping tracks");
+                    return false;
+                }
 
-            //void SetNComponents(unsigned nComps);
-            bool DoClustering();
-            bool Run();
+                KTDEBUG(otchlog, "Track clustering complete");
+                fDoneSignal();
 
+                return true;
+            }
 
-        private:
-            bool OverlapClustering();
-            bool DoTheyOverlap(KTProcessedTrackData& track1, KTProcessedTrackData& track2);
-            bool DoTheyCross(KTProcessedTrackData& track1, KTProcessedTrackData& track2);
-            const void CombineTracks(const KTProcessedTrackData& track1, KTProcessedTrackData& track2);
-            bool FindMatchingTracks();
-            void EmitTrackCandidates();
+            template<typename T>
```

Comment:
For all of these templated functions:
* Please move the function implementations outside of the class definition.  You can put it in this file, below the class definition.
* Please use a more descriptive type name than `T`.